climbing_stairs/climbing_stairs.py

```diff
@@ -2,13 +2,30 @@
 
 import sys
 
+"""
+# least steps (didn't read the instructions correctly)
 def climbing_stairs(n, cache=None):
-  pass 
+    threes = int(n / 3)
+    left_over = n % 3
+    twos = int(left_over / 2)
+    ones = left_over % 2
+    return threes + twos + ones
+"""
 
 
+def climbing_stairs(n, cache=None):
+    if n == 0:
+        return 1
+    elif n == 1 or n == 2:
+        return n
+    else:
+        return climbing_stairs(n-1) + climbing_stairs(n-2) + climbing_stairs(n-3)
```

Comment:
Nice recursive implementation. If you can tell me the complexity of this solution you're golden! (Hint Hint)